### DISCUSSION TITLE
Loading data set, using only the first card, memory overflow

### DISCUSSION THREAD

**Discussion by WillWillWong** [open-compass/opencompass#967] [Q&A]
When I was trying to reason the 13b model, I loaded the data set using only one card to run. My environment is 2*3090, and there are the following problems.torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 136.00 MiB. GPU 0 has a total capacity of 23.48 GiB of which 124.81 MiB is free. Including non-PyTorch memory, this process has 23.31 GiB memory in use. Of the allocated memory 23.06 GiB is allocated by PyTorch, and 2.76 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation. See documentation for Memory Management (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables).Loading checkpoint shards: 0%| | 0/3 [00:00<?, ?it/s]
Loading checkpoint shards: 33%|███▎ | 1/3 [00:04<00:09, 4.91s/it]
Loading checkpoint shards: 67%|██████▋ | 2/3 [00:09<00:04, 4.95s/it]
Loading checkpoint shards: 67%|██████▋ | 2/3 [00:11<00:05, 5.97s/it]
In fact, he only used the first card,
![image](https://github.com/open-compass/opencompass/assets/104180186/1ac6a9bc-ef63-42cb-99a0-1588e407a4a4)
![image](https://github.com/open-compass/opencompass/assets/104180186/83cd56d6-2244-4b06-afed-0af0bd1ffc98)

**Comment by bittersweet1999** [comment]
Besides --num-gpus, you also need to write 2gpus in your model cfg, just like below: 
```
models = [
    # LLaMA 13B
    dict(
        type=HuggingFaceCausalLM,
        abbr='llama-13b-hf',
        path="huggyllama/llama-13b",
        tokenizer_path='huggyllama/llama-13b',
        tokenizer_kwargs=dict(padding_side='left',
                              truncation_side='left',
                              use_fast=False,
                              ),
        max_out_len=100,
        max_seq_len=2048,
        batch_size=8,
        model_kwargs=dict(device_map='auto'),
        batch_padding=False, # if false, inference with for-loop without batch padding
        run_cfg=dict(num_gpus=2, num_procs=1),
    )
]
```
see in the https://github.com/open-compass/opencompass/blob/main/configs/models/hf_llama/hf_llama2_13b_chat.py

**Reply by WillWillWong** [reply]
I'm reasoning directly from the model address, as you can see in the second picture

**Reply by WillWillWong** [reply]
![image](https://github.com/open-compass/opencompass/assets/104180186/3d2a926f-ecb7-4a63-b719-5f15f4ec2f77)
I also wrote the model configuration file, but the effect is the same when using it. The memory is still overflowing

**Reply by bittersweet1999** [reply]
Please try device_map='auto'

**Reply by WillWillWong** [reply]
You are right, I had to use cuda when testing on glm3 and didn't think it would work on llama. What is the reason.
